src/wasm/wasm.cpp

```diff
@@ -1028,7 +1028,7 @@ void RefAs::finalize() {
       type = Type(value->type.getHeapType(), NonNullable);
       break;
     case RefAsFunc:
-      type = Type::funcref;
+      type = Type(HeapType::func, NonNullable);
```

Comment:
`dataref` and `i31ref` are currently specified to be non-nullable, but maybe we should expand them similarly here to be robust against future changes and to make this more obviously correct and consistent to readers who may not remember which abbreviations are nullable.

Reply:
Yes, I think it makes sense to revisit that suggestion, since operating on `funcref` and friends instead of the underlying heap type is almost never what we want. There might be a performance impact, but we have lower hanging fruit when it comes to GC type system performance ;)

Reply:
Yes, that's the key thing, I think - we almost never want to work on `funcref` etc. They are mainly a risk for confusion atm...

Ok, let's land this for now and keep thinking about this stuff.